subsys/bootloader/bl_boot/bl_boot.c

```diff
@@ -174,13 +223,18 @@ void bl_boot(const struct fw_info *fw_info)
 		"   b   clear\n"
 		"out:\n"
 		"   dsb\n"
+#ifndef CONFIG_SB_DISABLE_SELF_R_X
```

Comment:
If you disable the jump to vector, r0 is not needed here so the line 211 can be commented out and in line 233 the r0 does not have to be on list. I suspect that impact on size is not that great, but code will get uglier.